tools/cli.js

```diff
@@ -24,14 +24,64 @@ const log = (msg) => {
   if (userInput) rl.write(userInput);
 };
 
+function completeByFields(input, object) {
+  return Object.keys(object)
+    .filter(c => !c.startsWith('_') && (!input || c.startsWith(input)));
+}
+
+function tryComplete(input, completer) {
+  const completions = completer._complete([input], 0)[0];
```

Comment:
As far as I understand the `completer` interface, `_complete` and `_help` are its public methods. Why do they start with underscores?